integreat_cms/cms/forms/push_notifications/push_notification_translation_form.py

```diff
@@ -50,3 +51,19 @@ def add_error_messages(self, request):
             logger.debug(
                 "PushNotificationTranslationForm submitted with errors: %r", self.errors
             )
+
+    def has_changed(self):
+        """
+        Return ``True`` if submitted data differs from initial data.
+        If the main language should be used as fallback for missing translations, this always return ``True``.
+
+        :return: Whether the form has changed
+        :rtype: bool
+        """
+        if (
+            hasattr(self.instance, "push_notification")
+            and self.instance.push_notification.mode
+            == push_notifications.USE_MAIN_LANGUAGE
+        ):
+            return True
+        return super().has_changed()
```

Comment:
Instead of iterating though all forms, setting the data manually and overwriting the save method, we can also overwrite the form's `has_changed()` method to convince the formset to save these forms as well.